optimade/server/models/baseinfo.py

```diff
@@ -1,18 +1,19 @@
-from typing import Dict, List
+from typing import Dict, List, Optional
 
-from pydantic import BaseModel, UrlStr
+from pydantic import BaseModel, UrlStr, Schema
 from .modified_jsonapi import Resource
 
 
 class BaseInfoAttributes(BaseModel):
-    api_version: str
+    api_version: str = Schema(...)
     available_api_versions: Dict[str, UrlStr]
     formats: List[str] = ["jsonapi"]
     entry_types_by_format: Dict[str, List[str]]
-    available_endpoints: List[str] = ["structure", "all", "info"]
+    available_endpoints: List[str] = Schema(...)
+    is_index: Optional[bool] = Schema(default=False, const=True)
 
 
 class BaseInfoResource(Resource):
-    id: str = "/"
-    type: str = "info"
-    attributes: BaseInfoAttributes
+    id: str = Schema(default="/", const=True)
```

Comment:
I really like these two uses of `Schema` and its `const` :+1:

Reply:
One issue arising from this is that the example server code (as it stands) skips default values, meaning `id` and `type` here are omitted from the response. It's a simple change (see below) to the server code, but we should probably do it within this PR rather than in general server refactoring.

https://github.com/ml-evs/optimade-python-tools/blob/af349fc5e1f8f9214b9404797977914c9242112d/optimade/server/main.py#L155-158